### PR TITLE
DEV-5148 add functionality around first payment date

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

Brings back the Django management command and serializer changes in DEV-5059. See https://news-revenue-hub.atlassian.net/browse/ITS-2534

There are some additional changes I needed to make here because first_payment_date is temporarily nullable. For all of these, I added a TODO with the ticket number where we'll clean this up.

#### Why are we doing this? How does it help us?

Incremental deployment of problematic field.

#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?

Yes.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

See above re: TODO notes.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

- https://news-revenue-hub.atlassian.net/browse/DEV-5148
- https://news-revenue-hub.atlassian.net/browse/DEV-5139
- https://news-revenue-hub.atlassian.net/browse/ITS-2534

#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

Yes, see ITS story.

### CODE REVIEW

File: apps/contributions/tests/test_views.py [hunk 1]
```diff
@@ -1476,6 +1479,9 @@ def monthly_contribution(
     ):
         then = datetime.datetime.now(datetime.timezone.utc) - datetime.timedelta(days=30)
         contribution = ContributionFactory(
+            # TODO(ck): remove this when first_payment_date becomes non-nullable
```
Comment (nrh-cklimas) [inline apps/contributions/tests/test_views.py:1482]: New

File: apps/contributions/tests/test_views.py [hunk 2]
```diff
@@ -1506,6 +1512,9 @@ def yearly_contribution(
     ):
         then = datetime.datetime.now(datetime.timezone.utc) - datetime.timedelta(days=365)
         contribution = ContributionFactory(
+            # TODO(ck): remove this when first_payment_date becomes non-nullable
```
Comment (nrh-cklimas) [inline apps/contributions/tests/test_views.py:1515]: New

File: spa/src/components/portal/ContributionsList/ContributionDetail/BillingDetails/BillingDetails.tsx [hunk 3]
```diff
@@ -29,7 +29,8 @@ export const INTERVAL_NAMES: Record<ContributionInterval, string> = {
 
 export function BillingDetails({ contribution, disabled }: BillingDetailsProps) {
   const formattedDate = Intl.DateTimeFormat(undefined, { day: 'numeric', month: 'long', year: 'numeric' }).format(
-    new Date(contribution.created)
+    // TODO in DEV-5138: use only first_payment_date
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContributionDetail/BillingDetails/BillingDetails.tsx:32]: New

File: spa/src/components/portal/ContributionsList/ContributionItem/ContributionItem.test.tsx [hunk 4]
```diff
@@ -81,11 +82,20 @@ describe('ContributionItem', () => {
   it("doesn't throw if the contribution has a malformed interval", () =>
     expect(() => tree({ contribution: { ...mockContribution, interval: 'bad' } as any })).not.toThrow());
 
-  it('shows the date the contribution was made', () => {
+  // TODO in DEV-5138: Only need to test first_payment_date
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContributionItem/ContributionItem.test.tsx:85]: New

File: spa/src/components/portal/ContributionsList/ContributionItem/ContributionItem.tsx [hunk 5]
```diff
@@ -71,7 +71,9 @@ export const ContributionItem = forwardRef<HTMLAnchorElement, ContributionItemPr
           <IntervalIcon aria-label={getFrequencyAdjective(contribution.interval)} />
         </IntervalIconContainer>
         <DateContainer>
-          <CreatedDate data-testid="created">{formatDate(contribution.created)}</CreatedDate>
+          <CreatedDate data-testid="first-payment-date">
+            {formatDate(contribution.first_payment_date ?? contribution.created)}
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContributionItem/ContributionItem.tsx:75]: New

File: spa/src/hooks/usePortalContributionList.ts [hunk 6]
```diff
@@ -28,9 +28,15 @@ export interface PortalContribution {
...
-   * Timestamp of when the contribution was created.
+   * Timestamp of when the contribution was created. This is *not* necessarily
+   * when the transaction occurred.
    */
   created: string;
+  /**
+   * Timestamp of when the first related payment occurred.
+   * TODO in DEV-5139: Make this string only
+   */
+  first_payment_date: string | null;
```
Comment (nrh-cklimas) [inline spa/src/hooks/usePortalContributionList.ts:39]: New

**Review by njharman — APPROVED** [review]

**Review by GuiMend — APPROVED** [review]

**Review by x110dc — APPROVED** [review]

**Pull request merged**